controllers/dscinitialization/dscinitialization_test.go

```diff
@@ -0,0 +1,233 @@
+			Expect(foundConfigMap.Data).To(Equal(map[string]string{"namespace": "existing-data"}))
+			Expect(foundConfigMap.Data).ToNot(Equal(map[string]string{"namespace": applicationNamespace}))
+		})
+
+		It("Should not update namespace if it exists", func() {
+			applicationName := "configmap-test"
+			anotherNamespace := "test-another-ns"
+
+			// when
+			desiredNamespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: anotherNamespace,
+				},
+			}
+			Expect(k8sClient.Create(ctx, desiredNamespace)).Should(Succeed())
+			createdNamespace := &corev1.Namespace{}
+			Eventually(namespaceExists(anotherNamespace, createdNamespace), timeout, interval).Should(BeTrue())
+
+			// given
+			desiredDsci := getNewInstance(applicationName)
+			Expect(k8sClient.Create(ctx, desiredDsci)).Should(Succeed())
+			foundDsci := &dsci.DSCInitialization{}
+			Eventually(dsciExists(applicationName, workingNamespace, foundDsci), timeout, interval).Should(BeTrue())
+
+			//then
+			foundApplicationNamespace := &corev1.Namespace{}
+			Eventually(namespaceExists(anotherNamespace, foundApplicationNamespace), timeout, interval).Should(BeTrue())
+			Expect(foundApplicationNamespace.Name).To(Equal(createdNamespace.Name))
+			Expect(foundApplicationNamespace.UID).To(Equal(createdNamespace.UID))
+		})
+	})
+})
+
+// cleanup utility func
+func cleanupResources() {
+	defaultNamespace := client.InNamespace(workingNamespace)
+	appNamespace := client.InNamespace(applicationNamespace)
+	Expect(k8sClient.DeleteAllOf(context.TODO(), &dsci.DSCInitialization{}, defaultNamespace)).ToNot(HaveOccurred())
+	Expect(k8sClient.DeleteAllOf(context.TODO(), &netv1.NetworkPolicy{}, appNamespace)).ToNot(HaveOccurred())
+	Expect(k8sClient.DeleteAllOf(context.TODO(), &corev1.ConfigMap{}, appNamespace)).ToNot(HaveOccurred())
+	Expect(k8sClient.DeleteAllOf(context.TODO(), &authv1.RoleBinding{}, appNamespace)).ToNot(HaveOccurred())
+}
+
+func getNewInstance(appName string) *dsci.DSCInitialization {
+	return &dsci.DSCInitialization{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DSCInitialization",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appName,
+			Namespace: workingNamespace,
+		},
+		Spec: dsci.DSCInitializationSpec{
+			ApplicationsNamespace: applicationNamespace,
+			Monitoring: dsci.Monitoring{
+				Namespace:       monitoringNamespace,
+				ManagementState: operatorv1.Managed,
+			},
+		},
+	}
+}
+
+func namespaceExists(ns string, obj client.Object) func() bool {
+	return func() bool {
+		err := k8sClient.Get(context.Background(), client.ObjectKey{Name: ns}, obj)
+		return err == nil
+	}
+}
+
+func objectExists(ns string, name string, obj client.Object) func() bool {
+	return func() bool {
+		err := k8sClient.Get(context.Background(), client.ObjectKey{Name: ns, Namespace: name}, obj)
+		return err == nil
+	}
+}
+
+func dsciExists(ns string, name string, dsciObj *dsci.DSCInitialization) func() bool {
```

Comment:
I think this would be better reflecting actual assertion if we name it `dscInitializationIsReady`